TestDeadline fails with go 1.21.3

```  
command: go test -buildmode pie -compiler gc -ldflags " -X github.com/dlclark/regexp2/version=1.10.0 -extldflags '-Wl,-z,relro -Wl,--as-needed 
 -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -specs=/usr/lib/rpm/redh
at/redhat-package-notes  '"
      testing: github.com/dlclark/regexp2
github.com/dlclark/regexp2
--- FAIL: TestDeadline (0.00s)
    --- FAIL: TestDeadline/1ms (0.00s)
        fastclock_test.go:31: deadline (1ms) expired too soon (after 1.067327ms)
FAIL
exit status 1
FAIL    github.com/dlclark/regexp2      1.651s
```

This is still an issue in Rawhide (golang-1.24.3-4.fc43):

```
Executing(%check): /bin/sh -e /var/tmp/rpm-tmp.RC3RJq
+ umask 022
+ cd /builddir/build/BUILD/golang-github-dlclark-regexp2-1.11.5-build
+ cd regexp2-1.11.5
+ GO_LDFLAGS=' -X github.com/dlclark/regexp2/version=1.11.5'
+ GO_TEST_FLAGS='-buildmode pie -compiler gc'
+ GO_TEST_EXT_LD_FLAGS='-Wl,-z,relro -Wl,--as-needed  -Wl,-z,pack-relative-relocs -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-hardened-ld-errors -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -specs=/usr/lib/rpm/redhat/redhat-package-notes  '
+ go-rpm-integration check -i github.com/dlclark/regexp2 -b /builddir/build/BUILD/golang-github-dlclark-regexp2-1.11.5-build/regexp2-1.11.5/_build/bin -s /builddir/build/BUILD/golang-github-dlclark-regexp2-1.11.5-build/regexp2-1.11.5/_build -V 1.11.5-1.fc43 -p /builddir/build/BUILD/golang-github-dlclark-regexp2-1.11.5-build/BUILDROOT -g /usr/share/gocode -r '.*example.*'
Testing    in: /builddir/build/BUILD/golang-github-dlclark-regexp2-1.11.5-build/regexp2-1.11.5/_build/src
         PATH: /builddir/build/BUILD/golang-github-dlclark-regexp2-1.11.5-build/regexp2-1.11.5/_build/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/sbin
       GOPATH: /builddir/build/BUILD/golang-github-dlclark-regexp2-1.11.5-build/regexp2-1.11.5/_build:/usr/share/gocode
  GO111MODULE: off
      command: go test -buildmode pie -compiler gc -ldflags " -X github.com/dlclark/regexp2/version=1.11.5 -extldflags '-Wl,-z,relro -Wl,--as-needed  -Wl,-z,pack-relative-relocs -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-hardened-ld-errors -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -specs=/usr/lib/rpm/redhat/redhat-package-notes  '"
      testing: github.com/dlclark/regexp2
github.com/dlclark/regexp2
--- FAIL: TestDeadline (0.00s)
    --- FAIL: TestDeadline/1ms (0.00s)
        fastclock_test.go:32: deadline (1ms) expired too soon (after 1.078311ms)
    --- FAIL: TestDeadline/100µs (0.00s)
        fastclock_test.go:32: deadline (100µs) expired too soon (after 1.08281ms)
```

--- a/fastclock_test.go
+++ b/fastclock_test.go
@@ -14,10 +14,9 @@ func init() {
 }
 func TestDeadline(t *testing.T) {
 	for _, delay := range []time.Duration{
-		clockPeriod / 10,
-		clockPeriod,
-		clockPeriod * 5,
 		clockPeriod * 10,
+		clockPeriod * 50,
+		clockPeriod * 100,
 	} {
 		delay := delay // Make copy for parallel sub-test.
 		t.Run(fmt.Sprint(delay), func(t *testing.T) {

